simulator/infra/ports/t/unit_test.cpp

```diff
@@ -132,7 +200,42 @@ int main( int argc, char** argv)
 
 
 namespace ports {
-    namespace testing {
+    namespace testing {  
+
+        bool Logic::check_readiness( uint64 cycle, int check_code, bool is_ready)
+        {
+            assert( check_code >= 0 && check_code <= 3);
```

Comment:
It contradicts with a purpose of enumeration types which allow us to abstract from integer values. Could you please rewrite this without numbers?

Reply:
I can turn "int" of check_code into DataCode and ReadinessCode (enum ReadinessCode) in check_data and check_readiness respectively. If you are talking about assert I will handle it.

Reply:
That’s even worse, it will blow mind if person who would decide to call the function to add a new test case.. There should be 2 parameters of DataCode and Readiness code.

Reply:
I still don't get it. Should I avoid indexing in `data_logic` and `readiness_logic` through check_code which takes one of respective enums? I know it's not easy enough to add new case like DATA_FROM_C or IS_C_READY because it's necessary to keep in mind the way indexing works, but we benefit from simple implementation of check methods, do not we?